core/dbt/task/run.py

```diff
@@ -382,17 +382,18 @@ def defer_to_manifest(self, selected_uids: AbstractSet[str]):
                 'manifest to defer from!'
             )
         self.manifest.merge_from_artifact(
+            adapter=adapter,
             other=deferred_manifest,
             selected=selected_uids,
         )
         # TODO: is it wrong to write the manifest here? I think it's right...
         self.write_manifest()
 
     def before_run(self, adapter, selected_uids: AbstractSet[str]):
-        self.defer_to_manifest(selected_uids)
         with adapter.connection_named('master'):
             self.create_schemas(adapter, selected_uids)
             self.populate_adapter_cache(adapter)
+            self.defer_to_manifest(adapter, selected_uids)
```

Comment:
ok - i am seeing this now, and you can ignore my comment about caching above. Very clever, very cool. Leaving that comment above _parce que je suis impressionné_